chess/game.go

```diff
@@ -0,0 +1,52 @@
+package chess
+
+// A Game contains the history of a chess game.
+type Game struct {
+	Positions []Position
+	Moves     []Move
+
+	// The number of half moves since the last capture or pawn advance.
+	HalfMoveClock int
+
+	// The full move number for the next move.
+	FullMoveNumber int
```

Comment:
nit: fragments are probably fine

```suggestion
	// Half moves since the last capture or pawn advance.
	HalfMoveClock int

	// Full move number for the next move.
	FullMoveNumber int
```